ext/RMagick/rmimage.c

```diff
@@ -9762,11 +10740,18 @@ Image_pixel_color(int argc, VALUE *argv, VALUE self)
     if (!set)
     {
         exception = AcquireExceptionInfo();
-        old_color = *GetVirtualPixels(image, x, y, 1, 1, exception);
+        old_pixel = GetVirtualPixels(image, x, y, 1, 1, exception);
         CHECK_EXCEPTION()
 
         (void) DestroyExceptionInfo(exception);
 
+#if defined(IMAGEMAGICK_7)
+        old_color.red   = GetPixelRed(image, old_pixel);
+        old_color.green = GetPixelGreen(image, old_pixel);
+        old_color.blue  = GetPixelBlue(image, old_pixel);
+        old_color.alpha = GetPixelAlpha(image, old_pixel);
```

Comment:
For `IM6` there is already a define called `GetPixelRed` but we cannot use that here because it has it does not have the `image` argument.

Reply:
Because the image contains information at which index in the pointer the specified channel is. If the image is for example `RGB` the index for `R=0`, `G=1` and `B=2`. And that needs to be used as the offset from the `old_pixel` pointer. `B` would be `*(pointer+2)`. And if the order would be `GBR` the index would become `1` for `B`. For ImageMagick 6 the order is fixed instead.

Reply:
I see. Thanks for the explanation!